### PR TITLE
Hosted zone Particle

### PR DESCRIPTION
**Pull request by brianhlee**
has the changes for security group PR because I forgot to branch for that one. I'll do it right next time

### CODE REVIEW

File: examples/particle/aws/route53/use_route53/example_route53_hosted_zone.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+particle_definition = {
+    "pcf_name": "pcf_hosted_zone",
+    "flavor": "route53_hosted_zone",
+    "aws_resource": {
+        "Name": "www.hoooooos.com.",
+        "VPC": {
+            "VPCRegion": "us-east-1",
+            "VPCId": "vpc-12345"
+        },
+        "CallerReference": "werhasdkfboi12hasdfak",
```
Comment (anovis) [inline examples/particle/aws/route53/use_route53/example_route53_hosted_zone.py]: make random string

File: pcf/particle/aws/route53/route53_hosted_zone.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pcf.core.aws_resource import AWSResource
+from pcf.core import State
+from pcf.util import pcf_util
+
+
+class Route53HostedZone(AWSResource):
```
Comment (anovis) [inline pcf/particle/aws/route53/route53_hosted_zone.py]: lets just call it HostedZone. and it would sit in the route53 folder.

**Review by anovis — CHANGES_REQUESTED** [review]
minor changes, but otherwise looks good!

**Comment by davidyum** [comment]
Are there ways to add tags for hosted zone w/ boto3? 
Atleast in c1 I think they were enforcing tags at some point for hosted zone, not sure if its high priority outside but might be good to have
@bhl5kj

**Comment by brianhlee** [comment]
> Are there ways to add tags for hosted zone w/ boto3?

> @bhl5kj
I will look into it

**Review by anovis — APPROVED** [review]

**Pull request merged**